assembly/src/parsers/ast/mod.rs

```diff
@@ -73,7 +70,7 @@ impl ProgramAst {
 /// A module can only have exported and local procedures, but no body.
 #[derive(Debug, Eq, PartialEq)]
 pub struct ModuleAst {
-    pub procedures: LocalProcMap,
+    pub procedures: Vec<ProcedureAst>,
```

Comment:
minor nit: let's rename this to `local_procs` to indicate that this list contains only local procedures (i.e., procedures declared in this file).